src/actions/LogActions.tsx

```diff
@@ -104,16 +129,15 @@ export function showClearLogsModal(): ThunkAction<Promise<void>> {
   }
 }
 
-export function clearAllLogs(): ThunkAction<Promise<void>> {
-  return async (_dispatch, _getState) => {
-    await clearLogs('activity')
-    await logWithType('activity', 'CLEARED ACTIVITY LOGS')
-    await clearLogs('info')
-    await logWithType('info', 'CLEARED INFO LOGS')
+export async function clearAllLogs(): Promise<void> {
+  try {
+    await Promise.all([clearLogs('activity'), logWithType('activity', 'CLEARED ACTIVITY LOGS'), clearLogs('info'), logWithType('info', 'CLEARED INFO LOGS')])
+  } catch (error) {
+    console.error('Error while clearing logs:', error)
   }
 }
 
-function prepareLogs(text: string): ThunkAction<Promise<void>> {
+export function saveLogOutput({ userMessage, send = true }: LogOutputOptions): ThunkAction<Promise<MultiLogOutput>> {
```

Comment:
The layering is all messed up in this routine. Let's rename this back to `getLogOutput`, and have it simply return `Promise<MultiLogOutput>`, without any sending, sanitizing, or anything else. Just read from disk, and assemble the payload.

Then, the modal can call `saveLogOutput`, do the sanitatization, and report to the user if the sanitatization fails. That way you don't have `throw new Error()` with emojis  & translated strings inside the error message - you don't need to `throw` in the first place if you are already inside the modal. Just show whatever UI you need directly.

Finally, your modal can call either `sendLog` on the on the `activity` and `info` properties, or put them in the share modal, depending on the button press.

Reply:
The implementation of `getLogOutput` has been modified to return a `Promise<MultiLogOutput>` without performing any sanitation or sending. `handleSend` is disabled if `SENSITIVE_KEY_REGEX` matches either `logs.activity.data`. or `log.info.data`.